### ISSUE TITLE
Allow colorized custom formatters by exposing TextWriterExtensions

### ISSUE THREAD

**Issue by ramondeklein** [dotnet/runtime#44582]
## Background and Motivation
.NET Core 5 supports using custom console formatters, which is great to customize the logging output. Using colors is very useful for console logging to highlight important events, but each custom formatter needs to implement this.

The BCL already has a very useful [TextWriterExtensions](https://github.com/dotnet/runtime/blob/master/src/libraries/Microsoft.Extensions.Logging.Console/src/TextWriterExtensions.cs) that implements colorized output, but it's internal and can't be used for custom formatters. Making this class public would make it much easier to implement a custom formatter that uses colors the correct way.

## Proposed API
Make `Microsoft.Extensions.Logging.Console.TextWriterExtensions` public or make the `AnsiParser` accessible.

**Comment by ramondeklein** [comment]
@ericstj I suggest adding the `area-Extensions-Logging` label.

**Comment by SingleAccretion** [comment]
Hmm, this is very closely related to #44513. I think it would be good to consider some API that would work uniformly for both `Console.WriteLine` and `TextWriter` use cases.

**Comment by maryamariyan** [comment]
Thanks @ramondeklein as @SingleAccretion recommended, because of #44513, we'd want to preferably first expose some commmon APIs which fit better into System.Console and remove logic out of Logging.Console accordingly. That is why we decided not to make color specific formatting and Ansi parsing APIs public in Logging.Console as it might not be the right place for it.

Until those APIs are available, what I would do for custom formatting that would support colors, is to use the template gist I shared in the comment [here](https://github.com/dotnet/runtime/pull/40221#issuecomment-722565510), which basically just duplicates the same [TextWriterExtensions](https://github.com/dotnet/runtime/blob/master/src/libraries/Microsoft.Extensions.Logging.Console/src/TextWriterExtensions.cs) code you shared above.
 
If any such TextWriterExtensions APIs are needed to become public I think they should live in System.Console instead not in Logging.Console. 

cc @eerhardt perhaps we could dupe this as part of #44513 as well?
